fdbserver/KeyValueStoreRocksDB.actor.cpp

```diff
@@ -57,6 +70,71 @@ const StringRef ROCKSDB_READRANGE_NEWITERATOR_HISTOGRAM = LiteralStringRef("Rock
 const StringRef ROCKSDB_READVALUE_GET_HISTOGRAM = LiteralStringRef("RocksDBReadValueGet");
 const StringRef ROCKSDB_READPREFIX_GET_HISTOGRAM = LiteralStringRef("RocksDBReadPrefixGet");
 
+rocksdb::ExportImportFilesMetaData getMetaData(const CheckpointMetaData& checkpoint) {
+	rocksdb::ExportImportFilesMetaData metaData;
+	metaData.db_comparator_name = checkpoint.rocksCF.get().dbComparatorName;
+
+	if (checkpoint.rocksCF.present()) {
+		for (const LiveFileMetaData& fileMetaData : checkpoint.rocksCF.get().sstFiles) {
+			rocksdb::LiveFileMetaData liveFileMetaData;
+			liveFileMetaData.size = fileMetaData.size;
+			liveFileMetaData.name = fileMetaData.name;
+			liveFileMetaData.file_number = fileMetaData.file_number;
+			liveFileMetaData.db_path = fileMetaData.db_path;
+			liveFileMetaData.smallest_seqno = fileMetaData.smallest_seqno;
+			liveFileMetaData.largest_seqno = fileMetaData.largest_seqno;
+			liveFileMetaData.smallestkey = fileMetaData.smallestkey;
+			liveFileMetaData.largestkey = fileMetaData.largestkey;
+			liveFileMetaData.num_reads_sampled = fileMetaData.num_reads_sampled;
+			liveFileMetaData.being_compacted = fileMetaData.being_compacted;
+			liveFileMetaData.num_entries = fileMetaData.num_entries;
+			liveFileMetaData.num_deletions = fileMetaData.num_deletions;
+			liveFileMetaData.temperature = static_cast<rocksdb::Temperature>(fileMetaData.temperature);
+			liveFileMetaData.oldest_blob_file_number = fileMetaData.oldest_blob_file_number;
+			liveFileMetaData.oldest_ancester_time = fileMetaData.oldest_ancester_time;
+			liveFileMetaData.file_creation_time = fileMetaData.file_creation_time;
+			liveFileMetaData.file_checksum = fileMetaData.file_checksum;
+			liveFileMetaData.file_checksum_func_name = fileMetaData.file_checksum_func_name;
+			liveFileMetaData.column_family_name = fileMetaData.column_family_name;
+			liveFileMetaData.level = fileMetaData.level;
```

Comment:
Use a C++20 designated initializer list instead:

https://godbolt.org/z/17rr4YG47

That way, it'll fail to compile if RocksDB adds a field to the struct.

Reply:
Thanks for the tip! I like this new feature!

I played with it, but unfortunately, if fails because rocksdb::LiveFileMetaData has a `user provided constructor`.

https://godbolt.org/z/17rr4YG47

Let me know if I missed anything.